l1-contracts/test/Outbox.t.sol

```diff
@@ -1,135 +1,254 @@
 // SPDX-License-Identifier: Apache-2.0
-// Copyright 2023 Aztec Labs.
+// Copyright 2024 Aztec Labs.
 pragma solidity >=0.8.18;
 
 import {Test} from "forge-std/Test.sol";
-import {IOutbox} from "../src/core/interfaces/messagebridge/IOutbox.sol";
 import {Outbox} from "../src/core/messagebridge/Outbox.sol";
-import {Registry} from "../src/core/messagebridge/Registry.sol";
+import {IOutbox} from "../src/core/interfaces/messagebridge/IOutbox.sol";
 import {Errors} from "../src/core/libraries/Errors.sol";
 import {DataStructures} from "../src/core/libraries/DataStructures.sol";
-import {MessageBox} from "../src/core/libraries/MessageBox.sol";
+import {Hash} from "../src/core/libraries/Hash.sol";
+import {NaiveMerkle} from "./merkle/Naive.sol";
+import {MerkleTestUtil} from "./merkle/TestUtil.sol";
 
 contract OutboxTest is Test {
-  Registry internal registry;
-  Outbox internal outbox;
-  uint256 internal version = 0;
+  using Hash for DataStructures.L2ToL1Msg;
 
-  event MessageAdded(bytes32 indexed entryKey);
-  event MessageConsumed(bytes32 indexed entryKey, address indexed recipient);
+  address internal constant ROLLUP_CONTRACT = address(0x42069123);
+  address internal constant NOT_ROLLUP_CONTRACT = address(0x69);
```

Comment:
We don't really need a `not` value, we can do the fuzz, and then use `vm.assume` to make sure they are not the same if we really want to check them.